common/src/main/java/io/druid/collections/BlockingPool.java

```diff
@@ -72,17 +80,53 @@ public BlockingPool(
           @Override
           public void close() throws IOException
           {
-            if (!objects.offer(theObject)) {
-              log.error("WTF?! Queue offer failed, uh oh...");
+            offer(theObject);
```

Comment:
For avoiding the mistake to forget handling the case when ```offer()``` is failed. This case means there is a bug in returning unused resources. We are now simply logging an error message, but I think it should be improved.